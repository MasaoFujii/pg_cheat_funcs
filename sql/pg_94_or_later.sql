CREATE EXTENSION pg_cheat_funcs;

\pset null '(null)'

CREATE TABLE test_lsn (lsn pg_lsn);
INSERT INTO test_lsn VALUES ('0/A'), ('0/1000'), ('DA3/15A4D10'), ('E4/A0422B68'), ('447/F6D166E8');
SELECT min(lsn), max(lsn) FROM test_lsn;

SELECT pg_file_fsync('global');
SELECT pg_file_fsync('global/pg_control');

BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
SELECT pg_refresh_snapshot();
SELECT 1;
SELECT pg_refresh_snapshot();
COMMIT;

DROP EXTENSION pg_cheat_funcs;
