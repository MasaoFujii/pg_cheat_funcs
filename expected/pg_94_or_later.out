CREATE EXTENSION pg_cheat_funcs;
\pset null '(null)'
CREATE TABLE test_lsn (lsn pg_lsn);
INSERT INTO test_lsn VALUES ('0/A'), ('0/1000'), ('DA3/15A4D10'), ('E4/A0422B68'), ('447/F6D166E8');
SELECT min(lsn), max(lsn) FROM test_lsn;
 min |     max     
-----+-------------
 0/A | DA3/15A4D10
(1 row)

SELECT pg_file_fsync('global');
 pg_file_fsync 
---------------
 
(1 row)

SELECT pg_file_fsync('global/pg_control');
 pg_file_fsync 
---------------
 
(1 row)

BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;
SELECT pg_refresh_snapshot();
 pg_refresh_snapshot 
---------------------
 
(1 row)

SELECT 1;
 ?column? 
----------
        1
(1 row)

SELECT pg_refresh_snapshot();
 pg_refresh_snapshot 
---------------------
 
(1 row)

COMMIT;
DROP EXTENSION pg_cheat_funcs;
